src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -1324,18 +1324,131 @@ public static unsafe int SequenceCompareTo(ref byte first, int firstLength, ref
+                    if (matches == 0xFFFF)
+                    {
+                        // All matched
+                        goto Equal;
+                    }
+                Difference:
+                    // Invert matches to find differences
+                    int differences = ~matches;
+                    offset = (IntPtr)((int)(byte*)offset + BitOps.TrailingZeroCount(differences));
+
+                    int result = Unsafe.AddByteOffset(ref first, offset).CompareTo(Unsafe.AddByteOffset(ref second, offset));
+                    Debug.Assert(result != 0);
+
+                    return result;
+                }
+            }
+            else if (Sse2.IsSupported)
+            {
+                if ((byte*)nLength >= (byte*)Vector128<byte>.Count)
+                {
+                    nLength -= Vector128<byte>.Count;
+                    int matches;
+                    while ((byte*)nLength > (byte*)offset)
+                    {
+                        matches = Sse2.MoveMask(Sse2.CompareEqual(LoadVector128(ref first, offset), LoadVector128(ref second, offset)));
+                        if (matches == 0xFFFF)
+                        {
+                            // All matched
+                            offset += Vector128<byte>.Count;
+                            continue;
+                        }
+
+                        goto Difference;
+                    }
+                    // Move to Vector length from end for final compare
+                    offset = nLength;
+                    matches = Sse2.MoveMask(Sse2.CompareEqual(LoadVector128(ref first, offset), LoadVector128(ref second, offset)));
+                    if (matches == 0xFFFF)
+                    {
+                        // All matched
+                        goto Equal;
+                    }
+                Difference:
+                    // Invert matches to find differences
+                    int differences = ~matches;
+                    offset = (IntPtr)((int)(byte*)offset + BitOps.TrailingZeroCount(differences));
+
+                    int result = Unsafe.AddByteOffset(ref first, offset).CompareTo(Unsafe.AddByteOffset(ref second, offset));
+                    Debug.Assert(result != 0);
+
+                    return result;
+                }
+            }
+            else if (Vector.IsHardwareAccelerated)
+            {
+                if ((byte*)nLength > (byte*)Vector<byte>.Count)
+                {
+                    nLength -= Vector<byte>.Count;
+                    while ((byte*)nLength > (byte*)offset)
+                    {
+                        if (LoadVector(ref first, offset) != LoadVector(ref second, offset))
+                        {
+                            goto NotEqual;
+                        }
+                        offset += Vector<byte>.Count;
+                    }
+                    goto NotEqual;
```

Comment:
Why do we need this `goto NotEqual`?

Reply:
Previous implementation; it moves to the bytewise examination.

Could either rename the jump point `NotEqual:` => `CheckByteWise:` or set `nLength` back e.g. `nLength = minLength;` so we can fall through?

Reply:
Renamed to `BytewiseCheck`

Reply:
Renamed to `BytewiseCheck` as `NotEqual` didn't really make sense.